ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/JsonQueryStreamResponseWriter.java

```diff
@@ -76,7 +112,23 @@ public QueryStreamResponseWriter writeRow(
     if (keyValue.value() == null) {
       LOG.warn("Dropped tombstone. Not currently supported");
     } else {
-      writeBuffer(ServerUtils.serializeObject(keyValue.value().values()));
+      if (mediaType.equals(KsqlMediaType.KSQL_V1_PROTOBUF.mediaType())) {
```

Comment:
Do I understand correctly that you merged `JsonQueryStreamResponseWriter` and `ProtobufQueryStreamResponseWriter` since the last review review round and the motivation was to reduce the code duplication as @agavra asked?

There are some differences between `ProtobufQueryStreamResponseWriter` and ` JsonQueryStreamResponseWriter` that were not ported, specifically in methods `writeContinuationToken`, `writeError`, `writeConsistencyToken`, `writeCompletionMessage`, `writeLimitMessage`. Why are they no longer necessary?

Reply:
FWIW, I think that it would be better to have format-specific implementations in different classes rather than having conditions in every method. W could reduce code duplication by:
* moving default implementations to the interface or abstract class
* moving format-specfic serialization to a separate class or method.

For completness, we should not necessarily block this PR on it; it is fine with me to just create a follow-up issue.

Reply:
addressed